rfcs/20240819-generic-vendor/README.md

```diff
@@ -0,0 +1,78 @@
+# Enable Generic GPU Vendor
+
+## Motivation
+
+oneDNN project is part of the Unified Acceleration Foundation (UXLF) ecosystem,
+and as part of the commitment to the UXLF strategy oneDNN will introduce primitive
+implementations that can work with multiple different GPU and accelerator vendors.
+
+
+## Proposal
+
+In this RFC the primitive implementations will be called "generic".
+The generic implementations (kernels) will be implemented in SYCL. oneDNN already has a
+set of such kernels that have extensive coverage and are already enabled for
+NVIDIA and AMD vendors. This proposal is about enabling the kernels for a "generic"
+vendor that could be used to run the kernels on different hardware.
+
+SYCL specification defines the following device types:
+* CPU
+* GPU
+* Accelerator
+
+Each of the aforementioned device types is independent from the others and therefore
+SYCL devices have to be created specifically for particular types.
+
+oneDNN supports two engine kinds: CPU and GPU. According to the UXFL strategy the
+generic kernels are expected to run on different GPUs and accelerators.
```

Comment:
Do you know if there are kernel support differences between GPUs and Accelerators in SYCL (e.g. memory model, functionalities not mandated, ...)?

Reply:
Generic SYCL is meant to be a common denominator for all devices so I don't think there should be any differences in the SYCL programming model. However, different devices may have different capabilities and therefore some of the SYCL features may or may not be available for them. For example, USM memory model may not be supported by some devices so the users have to check the corresponding device aspect to query the capability prior to allocating USM memory. 
We should also be careful with those cases inside the library, e.g. we always allocate a SYCL buffer for the scratchpad, which is always supported, but if we decide to use USM memory for it we will need to do that properly.